### PR TITLE
capabilities blacklist

### LINKED ISSUE #1074: Make robot capabilities use a blacklist instead of a whitelist
### Acceptance criteria

- [ ] Item A
- [ ] Item B
- [ ] Item C

### PR DESCRIPTION
**Pull request by jordanzliu**
### Description

make robotcapabilities use a blacklist instead of a whitelist, to improve performance

### Testing Done

existing robot capabiltiies tests pass, tests modified to use blacklist

### Resolved Issues

resolves #1074 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by kelvinkoon — DISMISSED** [review]
LGTM!

File: src/software/ai/hl/stp/stp_tactic_assignment_test.cpp [hunk 1]
```diff
@@ -358,9 +358,11 @@ TEST_F(STPTacticAssignmentTest,
     // that doesn't even though the former has lower cost
     Team friendly_team(Duration::fromSeconds(0));
     // this robot has no capabilities
+    using RobotCapabilities::Capability;
     Robot robot_0(0, Point(0.1, 0.1), Vector(), Angle::zero(), AngularVelocity::zero(),
                   Timestamp::fromSeconds(0), 10,
-                  std::set<RobotCapabilities::Capability>{});
+                  std::set<RobotCapabilities::Capability>{Capability::Move, Capability::Kick,
+                                                                                Capability::Chip, Capability::Dribble});
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/stp_tactic_assignment_test.cpp]: nit: can this be allCapabilities()?

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Looks good, just a nit

**Review by garethellis0 — COMMENTED** [review]
LGTM pending Jon's comment

File: src/software/ai/hl/stp/stp_tactic_assignment_test.cpp (hunk 1 above)
Comment (garethellis0) [reply src/software/ai/hl/stp/stp_tactic_assignment_test.cpp]: Agreed, this would be nice.

**Review by MathewMacDougall — COMMENTED** [review]
LGTM pending Jon's comment

**Review by matthewberends — COMMENTED** [review]
Nothing to add on top of the existing comments.

**Review by MathewMacDougall — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]
:+1:

**Pull request merged**